rclc/include/rclc/executor_handle.h

```diff
@@ -0,0 +1,194 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#ifndef RCLC__EXECUTOR_HANDLE_H_
+#define RCLC__EXECUTOR_HANDLE_H_
+
+#if __cplusplus
+extern "C"
+{
+#endif
+
+#include <rcl/rcl.h>
+
+/// TODO (jst3si) Where is this defined? - in my build environment this variable is not set.
+// #define ROS_PACKAGE_NAME "rclc"
+
+/// Enumeration for timer, subscription, guard conditions etc to be waited on.
+typedef enum
+{
+  SUBSCRIPTION,
+  GUARD_CONDITION,  // not implemented yet
+  TIMER,
+  CLIENT,  // not implemented yet
+  SERVICE,  // not implemented yet
+  NONE
+} rclc_executor_handle_type_t;
+
+/// Enumeration for invocation type. ON_NEW_DATA calls a callback only when new data is available
+/// ALWAYS calls the callback always, even if no data is available (e.g. for type FUNCTION_CALL)
+typedef enum
+{
+  ON_NEW_DATA,
+  ALWAYS
+} rclc_executor_handle_invocation_t;
+
+/// Type defintion for callback function.
+typedef void (* rclc_callback_t)(const void *);
+
+/// Container for a handle.
+typedef struct
+{
+  /// Type of handle
+  rclc_executor_handle_type_t type;
+  /// When to execute callback
+  rclc_executor_handle_invocation_t invocation;
+  union {
+    /// Storage of subscription pointer
+    rcl_subscription_t * subscription;
+    /// Storage of timer pointer
+    rcl_timer_t * timer;
+    // rcl_service_t
+    // rcl_client_t
+    // rcl_guard_condition_t
+  };
+  /// Storage of data, which holds the message of a subscription, service, etc.
+  void * data;
+  /// Storage for callback for subscription
+  rclc_callback_t callback;
+  /// Internal variable.
+  /**  Denotes the index of this handle in the correspoding wait_set entry.
+  *    (wait_set_subscriptions[index], wait_set_timers[index], ...
+  *    is in the range [0,executor.max_handles), initialization: executor_max_handles
+  *    because this value will never be assigned as an index in the wait_set.
+  */
+  size_t index;
+  /// Internal variable. Flag, which is true, if the handle is initialized and therefore initialized
+  bool initialized;
+  /// Interval variable. Flag, which is true, if new data is available from DDS queue
+  /// (is set after calling rcl_take)
+  bool data_available;
+} rclc_executor_handle_t;
+
+/// Information about total number of subscriptions, guard_conditions, timers, subscription etc.
+typedef struct
+{
+  /// Total number of subscriptions
+  size_t number_of_subscriptions;
+  /// Total number of guard conditions
+  size_t number_of_guard_conditions;
+  /// Total number of timers
+  size_t number_of_timers;
+  /// Total number of clients
+  size_t number_of_clients;
+  /// Total number of services
+  size_t number_of_services;
+  /// Total number of events
+  size_t number_of_events;
+} rclc_executor_handle_counters_t;
+
```

Comment:
```suggestion
```

Reply:
fixed in 54f0bbc54a33cc601764ab1d53fe3119d9e61c8b